service/history/transfer_queue_active_task_executor.go

```diff
@@ -843,33 +843,42 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 	// This path is usually taken when the parent is being reset and the reset point (i.e baseWorkflowInfo.LowestCommonAncestorEventId) is after the child was initiated.
-	baseWorkflowInfo := mutableState.GetBaseWorkflowInfo()
-	if mutableState.IsResetRun() && baseWorkflowInfo != nil && baseWorkflowInfo.LowestCommonAncestorEventId >= childInfo.InitiatedEventId {
-		childRunID, err := t.verifyChildWorkflow(ctx, mutableState, targetNamespaceEntry, attributes.WorkflowId)
-		if err != nil {
-			return err
-		}
-		if childRunID != "" {
-			childExecution := &commonpb.WorkflowExecution{
-				WorkflowId: childInfo.StartedWorkflowId,
-				RunId:      childRunID,
-			}
-			childClock := childInfo.Clock
-			// Child execution is successfully started, record ChildExecutionStartedEvent in parent execution
-			err = t.recordChildExecutionStarted(ctx, task, weContext, attributes, childRunID, childClock)
+	shouldTerminateAndStartChild := false
+	resetChildID := fmt.Sprintf("%s:%s", attributes.GetWorkflowType().Name, attributes.GetWorkflowId())
+	if mutableState.IsResetRun() {
+		baseWorkflowInfo := mutableState.GetBaseWorkflowInfo()
+		if baseWorkflowInfo != nil && baseWorkflowInfo.LowestCommonAncestorEventId >= childInfo.InitiatedEventId { // child was started before the reset point.
+			childRunID, err := t.verifyChildWorkflow(ctx, mutableState, targetNamespaceEntry, attributes.WorkflowId)
 			if err != nil {
 				return err
 			}
-			// NOTE: do not access anything related mutable state after this lock release
-			// release the context lock since we no longer need mutable state and
-			// the rest of logic is making RPC call, which takes time.
-			release(nil)
+			if childRunID != "" {
+				childExecution := &commonpb.WorkflowExecution{
+					WorkflowId: childInfo.StartedWorkflowId,
+					RunId:      childRunID,
+				}
+				childClock := childInfo.Clock
+				// Child execution is successfully started, record ChildExecutionStartedEvent in parent execution
+				err = t.recordChildExecutionStarted(ctx, task, weContext, attributes, childRunID, childClock)
+				if err != nil {
+					return err
+				}
+				// NOTE: do not access anything related mutable state after this lock release
+				// release the context lock since we no longer need mutable state and
+				// the rest of logic is making RPC call, which takes time.
+				release(nil)
 
-			parentClock, err := t.shardContext.NewVectorClock()
-			if err != nil {
-				return err
+				parentClock, err := t.shardContext.NewVectorClock()
+				if err != nil {
+					return err
+				}
+				return t.createFirstWorkflowTask(ctx, task.TargetNamespaceID, childExecution, parentClock, childClock)
+			}
+		} else {
+			// child was started after reset-point. But we need to first check if this child was recorded at the time of reset.
+			if resetChildInfo, ok := mutableState.GetExecutionInfo().GetChildrenInitializedPostResetPoint()[resetChildID]; ok {
```

Comment:
per our discussion: this check needs to be done even if `mutableState.IsResetRun()` is false to support continue as new case.

Reply:
ah! yes. Fixed it.